src/bin/sol-fbp-generator/type-store.c

```diff
@@ -384,6 +400,73 @@ parse_range_default_value(struct sol_json_scanner *s, struct option_description
 }
 
 static bool
+parse_rgb_default_value(struct sol_json_scanner *s, struct option_description *o)
+{
+    struct option_rgb_value *rgb;
+    struct sol_json_token tmp, key, value;
+    struct sol_str_slice key_slice;
+    char *value_data;
+
+    rgb = &o->default_value.rgb;
+    memset(rgb, 0, sizeof(struct option_rgb_value));
+
+    if (!sol_json_scanner_next(s, &tmp))
+        return false;
+    if (sol_json_token_get_type(&tmp) != SOL_JSON_TYPE_OBJECT_START)
+        return false;
+
+    for (;;) {
```

Comment:
There are macros to help you here. See `SOL_JSON_SCANNER_OBJECT_LOOP`. It's used in the OIC stuff if you want a few examples.